products/views.py

```diff
@@ -1,3 +1,73 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http      import JsonResponse
+from django.views     import View
+from django.db.models import Q
+
+from datetime       import datetime
+from .models        import Category, Subcategory, Product
+from users.models   import User, Host
+
+class ProductView(View): 
+    def get(self, request):
+        try: 
+            category    = request.GET.get('category',None)
+            subcategory = request.GET.get('subcategory',None)
+            region      = request.GET.get('region',None)
+            group       = request.GET.get('group',None)
+            price       = request.GET.get('price',None)
```

Comment:
`request.GET.get(, None)`의 경우에는 default로 `None`을 반환하게 되는데 아래 필터링 할때 문제 안생기나요?

` q_object &= Q(subcategory__category=None)` 괜찮은 건가요?

```suggestion
            category    = request.GET.get('category',None)
            subcategory = request.GET.get('subcategory',None)
            region      = request.GET.get('region',None)
            group       = request.GET.get('group',None)
            price       = request.GET.get('price',None)
```

Reply:
if category: 로 None값을 걸려주었기 때문에 괜찮은것 같습니다!